### PR TITLE
Fix guzzle issue with query string encoding

### LINKED ISSUE #1121: Email not returned when sigining into facebook using guzzle instead of curl
## Bug, feature or question?

when I swap out curl for guzzle and the user signs into Facebook the user profile will not return the email even though the email permission is granted by the user. This does not happen with the curl set up nor for google on the guzzle set up.

### Version and provider

v 3.0

### PR DESCRIPTION
**Pull request by chrisgraham**
| Q                        | A
| ------------------------ | ---
| Fixed Issues?            | Fixes #1121 
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  | No
| Minor: New Feature?      | No

Guzzle won't merge query strings passed via URL and via 'query', so do it for it.

### CODE REVIEW

File: src/HttpClient/Guzzle.php [hunk 1]
```diff
@@ -115,6 +115,10 @@ public function request($uri, $method = 'GET', $parameters = [], $headers = [],
     {
         $this->requestHeader = array_replace($this->requestHeader, (array) $headers);
 
+        if (($method == 'GET' || $method == 'DELETE') && (strpos($uri, '?') !== false)) {
```
Comment (ApacheEx) [inline src/HttpClient/Guzzle.php:118]: not sure - in fact query params can be in any kind of http method. Also, if there is such case - we probably just need to move query from the url to query as array.

**Review by ApacheEx — CHANGES_REQUESTED** [review]
what do you think for my comment?

**Comment by chrisgraham** [comment]
Yeah, that's fine, we can do that, it'll just involve a few different changes which I can do when the pull requests are settled.

**Comment by chrisgraham** [comment]
I've fixed all the cases of implicit query strings. Except for Facebook, which already is fixed in one of my other pull requests by coincidence, and I don't want to cause a merge conflict by making the same kind of change here.

File: src/Provider/LinkedIn.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ public function getUserProfile()
         ];
 
 
-        $response = $this->apiRequest('me?projection=(' . implode(',', $fields) . ')');
+        $response = $this->apiRequest('me', 'GET', [ 'projection' => '(' . implode(',', $fields) . ')' ]);
```
Comment (ApacheEx) [inline src/Provider/LinkedIn.php:58]: let's avoid this space after `[` and before `]`. Pls apply it below as well

**Review by ApacheEx — CHANGES_REQUESTED** [review]
a small remark about coding style :) hope, it's not a problem

File: src/Provider/LinkedIn.php (hunk 2 above)
Comment (chrisgraham) [reply src/Provider/LinkedIn.php:58]: I'm not offended, but actually I was trying to be consistent with existing code ;-).

```
./src/HttpClient/Guzzle.php: *      'headers' => [ 'User-Agent' => '..' ]
./src/HttpClient/Curl.php:        $this->curlOptions[CURLOPT_HEADERFUNCTION] = [ $this, 'fetchResponseHeader' ];
./src/HttpClient/Curl.php:            $this->curlOptions[ $opt ] = $value;
./src/HttpClient/Curl.php:            $this->responseHeader[ $key ] = $value;
./src/Provider/Yandex.php:        $response = $this->apiRequest($this->apiBaseUrl, 'GET', [ 'format' => 'json' ]);
./src/Provider/Steam.php: *       'keys'     => [ 'secret' => 'steam-api-key' ]
./src/Provider/Twitter.php: *       'keys'      => [ 'key' => '', 'secret' => '' ], // OAuth1 uses 'key' not 'id'
./src/Provider/Twitter.php:        $parameters = [ 'cursor' => '-1'] + $parameters;
./src/Provider/Twitter.php:            $parameters = [ 'user_id' => implode(',', $chunk) ];
./src/Provider/Google.php: *       'keys'     => [ 'id' => '', 'secret' => '' ],
./src/Provider/Google.php:                    . http_build_query(array_replace([ 'alt' => 'json', 'v' => '3.0' ], (array)$parameters));
./src/Provider/Facebook.php: *       'keys'     => [ 'id' => '', 'secret' => '' ],
./src/Provider/StackExchange.php: *       'keys'     => [ 'id' => '', 'secret' => '' ],
./src/Provider/StackExchange.php:        $this->apiRequestParameters = [ 'key' => $apiKey];
./src/Provider/StackExchange.php:        $response = $this->apiRequest('me', 'GET', [ 'site' => $site ]);
./src/Provider/MicrosoftGraph.php: *       'keys'     => [ 'id' => '', 'secret' => '' ],
./src/Provider/DeviantArt.php: *       'keys'     => [ 'id' => '', 'secret' => '' ],
./src/Provider/Tumblr.php:                    ? [ 'type' => 'text', 'body' => $status ]
./src/Provider/LinkedIn.php:        $response = $this->apiRequest('me', 'GET', [ 'projection' => '(' . implode(',', $fields) . ')' ]);
./src/Provider/LinkedIn.php:        $response = $this->apiRequest('emailAddress', 'GET', [ 'q' => 'members', 'projection' => '(elements*(handle~))' ]);
./src/Provider/Paypal.php: *       'keys'     => [ 'id' => '', 'secret' => '' ],
./src/Provider/Apple.php: *       'keys'     => [ 'id' => '', 'team_id' => '', 'key_id' => '', 'key_file' => '', 'key_content' => '' ],
./src/Provider/Apple.php:                $userProfile->displayName = join(' ', [ $userProfile->firstName, $userProfile->lastName ]);
./src/Adapter/AbstractAdapter.php:                $tokens[ $name ] = $this->getStoredData($name);
./src/Storage/Session.php:                    unset($tmp[ $k ]);
./src/Data/Parser.php:        return [ $birthday['year'], $birthday['month'], $birthday['day'] ];
./README.md:    'keys' => [ 'key' => 'your-twitter-consumer-key', 'secret' => 'your-twitter-consumer-secret' ]
./tests/Storage/SessionTest.php:            [ 'foo', 'bar' ],
./tests/Storage/SessionTest.php:            [  1234, 'bar' ],
./tests/Storage/SessionTest.php:            [ 'foo',  1234 ],
./tests/Storage/SessionTest.php:            [ 'Bonjour', '안녕하세요' ],
./tests/Storage/SessionTest.php:            [ 'ஹலோ'   , 'Γεια σας' ],
./tests/Storage/SessionTest.php:            [ 'array', [ 1, 2, 3 ] ],
./tests/Storage/SessionTest.php:            [ 'string',  json_encode($this) ],
./tests/Storage/SessionTest.php:            [ 'object', $this ],
./tests/Storage/SessionTest.php:            [ 'provider.token.request_token', '9DYPEJ&qhvhP3eJ!' ],
./tests/Storage/SessionTest.php:            [ 'provider.token.oauth_token', '80359084-clg1DEtxQF3wstTcyUdHF3wsdHM' ],
./tests/Storage/SessionTest.php:            [ 'provider.token.oauth_token_secret', 'qiHTi1znz6qiH3tTcyUdHnz6qiH3tTcyUdH3xW3wsDvV08e' ],
./examples/example_07/config.php:        // 'Yahoo'     => ['enabled' => true, 'keys' => [ 'key'  => '...', 'secret' => '...']],
./examples/example_07/config.php:        // 'Facebook'  => ['enabled' => true, 'keys' => [ 'id'   => '...', 'secret' => '...']],
./examples/example_07/config.php:        // 'Twitter'   => ['enabled' => true, 'keys' => [ 'key'  => '...', 'secret' => '...']],
./examples/example_07/config.php:        // 'Instagram' => ['enabled' => true, 'keys' => [ 'id'   => '...', 'secret' => '...']],
./examples/example_02.php:        'GitHub' => [ 
./examples/example_02.php:            'keys'    => [ 'id' => '', 'secret' => '' ], 
./examples/example_02.php:        'Google' => [ 
./examples/example_02.php:            'keys'    => [ 'id' => '', 'secret' => '' ],
./examples/example_02.php:        'Facebook' => [ 
./examples/example_02.php:            'keys'    => [ 'id' => '', 'secret' => '' ],
./examples/example_02.php:        'Twitter' => [ 
./examples/example_02.php:            'keys'    => [ 'key' => '', 'secret' => '' ],
./examples/example_05.php:    'keys' => [ 'id' => '', 'secret' => '' ],
./examples/example_01.php:    'keys' => [ 'id' => 'your-app-id', 'secret' => 'your-app-secret' ], // Your Github application credentials
./examples/example_03.php:    'keys' => [ 'id' => 'your-facebook-app-id', 'secret' => 'your-facebook-app-secret' ],
```

How about I make you a little deal... accept this pull request, but then I make another one that just fixes us to PSR-2 formatting consistently?

Comment (ApacheEx) [reply src/Provider/LinkedIn.php:58]: agree, this makes sense. I'll approve it - but it seems you need to fix a merge conflict.

**Review by ApacheEx — APPROVED** [review]
finally, CI is ✅

**Pull request merged**